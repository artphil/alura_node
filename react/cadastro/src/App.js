import './App.css';

import { Container, Typography } from '@mui/material';
import '@fontsource/roboto';

import FormularioCadastro from './components/FormularioCadastro';

function App() {
  return (
    <Container maxWidth='sm' component='article'>
      <Typography 
      variant='h3'
      component='h1'
      align='center'
      >
      Formulário de cadastro
      </Typography>


      <h1>Formulário de cadastro</h1>
      <FormularioCadastro aoEnviar={aoEnviar} validarCPF={validaCpf}/>
    </Container>
  );
}

function aoEnviar(dados){
  console.log(dados);
}

function validaCpf(cpf){
  if(cpf.length !== 11){
    return {valido:false, texto:'CPF deve ter 11 digitos.'};
  }
  return {valido:true, texto:''};
}

export default App;
