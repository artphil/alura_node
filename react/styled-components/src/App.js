import React, { useState } from "react";
import { ThemeProvider } from "styled-components";


import Cabecalho from "./Components/Cabecalho";
import Container from "./Components/Container";
import GlobalStyle from "./Components/GlobalStyled";
import SwitcherTema from "./Components/SwitcherTema";
import { BotaoTema } from "./Components/UI";
import { temaClaro, temaEscuro } from "./Components/UI/temas";

function App() {
  const [tema, setTema] = useState(true);

  const toggleTema = () => {
    setTema(!tema);
  };

  return (
    <ThemeProvider theme={tema ? temaClaro : temaEscuro}>
      <GlobalStyle />
      <BotaoTema onClick={toggleTema}>
        <SwitcherTema tema={tema} />
      </BotaoTema>
      <Cabecalho />
      <Container />
    </ThemeProvider>
  );
}

export default App;
