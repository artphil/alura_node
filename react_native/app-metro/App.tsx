import * as React from 'react';

import Estacoes from './src/telas/Estacoes';

export default function App() {
  return (<>
    <Estacoes />
  </>);
}


