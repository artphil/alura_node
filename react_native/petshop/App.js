import React from 'react';
import { StyleSheet } from 'react-native';
import 'intl';
import 'intl/locale-data/jsonp/pt-BR';

import Rotas from './src/Rotas';
import TelaPadrao from './src/componentes/TelaPadrao';

export default function App() {
  return <TelaPadrao>
    <Rotas />
  </TelaPadrao>;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
