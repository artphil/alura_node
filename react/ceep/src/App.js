import React, { Component } from "react";

import FormularioCadastro from "./componentes/FormularioCadastro";
import ListaDeNotas from "./componentes/ListaDeNotas";
import ListaDeCategorias from "./componentes/ListaDeCategorias";

import './assets/index.css';
import './assets/App.css';
import Categorias from "./dados/Categorias";
import ArrayDeNotas from "./dados/Notas";


class App extends Component {

  constructor() {
    super();
    this.notas = new ArrayDeNotas();
    this.categorias = new Categorias();
  }

  // criarNota(titulo, texto, categoria) {
  //   const novaNota = { titulo, texto, categoria };
  //   const novoArrayNotas = [...this.state.notas, novaNota];
  //   const novoEstado = { ...this.state, notas: novoArrayNotas }
  //   this.setState(novoEstado)
  // }

  // deletarNota(index) {
  //   let arrayNotas = this.state.notas;
  //   arrayNotas.splice(index, 1);
  //   this.setState({ notas: arrayNotas })
  // }

  // adicionarCategoria(nomeCategoria) {
  //   const novoArrayCategoria = [ ...this.state.categorias, nomeCategoria ]
  //   const novoEstado = { ...this.state, categorias:novoArrayCategoria };
  //   this.setState(novoEstado);
  // }

  render() {
    return (
      <section className="conteudo">
        <FormularioCadastro 
        categorias={this.categorias}
        notas={this.notas} 
        />
        <main className="conteudo-principal">
          <ListaDeCategorias 
          categorias={this.categorias} 
          />
          <ListaDeNotas
            notas={this.notas}
          />
        </main>
      </section>
    );
  }
}

export default App;
