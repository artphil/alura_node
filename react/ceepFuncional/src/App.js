import React, { useState } from "react";

import FormularioCadastro from "./componentes/FormularioCadastro";
import ListaDeCategorias from "./componentes/ListaDeCategorias";
import ListaDeNotas from "./componentes/ListaDeNotas";

import './assets/index.css';
import './assets/App.css';


function App() {

  const [notas, setNotas] = useState([])
  const [categorias, setCategorias] = useState([])

  const criarNota = (titulo, texto, categoria) => {
    const novaNota = { titulo, texto, categoria };
    const novoArrayNotas = [...notas, novaNota];
    setNotas(novoArrayNotas)
  }

  const deletarNota = (index) => {
    let arrayNotas = this.notas;
    arrayNotas.splice(index, 1);
    setNotas(arrayNotas)
  }

  const criarCategoria = (nome) => {
    const novoArrayCategorias = [...categorias, nome];
    setCategorias(novoArrayCategorias)
  }

  const deletarCategoria = (index) => {
    let arrayCategorias = this.Categorias;
    arrayCategorias.splice(index, 1);
    setCategorias(arrayCategorias)
  }

  return (
    <section className="conteudo">
      <FormularioCadastro
        criarNota={criarNota}
        categorias={categorias}
      />
      <main className="conteudo-principal">
        <ListaDeCategorias
          categorias={categorias}
          adicionarCategoria={criarCategoria}
          removerCategoria={deletarCategoria}
        />
        <ListaDeNotas
          notas={notas}
          apagarNota={deletarNota}
        />

      </main>
    </section>
  );
}

export default App;
