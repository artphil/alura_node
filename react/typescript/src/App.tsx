import React from 'react';
import './App.css';
import Home from './telas/Home';

function App() {
  return (
      <Home/>
  );
}

export default App;
