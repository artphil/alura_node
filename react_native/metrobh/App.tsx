/*
  Arquivo principal do programa
  Arquivo que é chamado pelo React e Expo no início da aplicação
  O nome e local não podem ser alterados sem modificar o 
  módulo do Expo
  Este arquivo está apenas redirecionando para o código que está em 
  src/index.tsx
*/
import App from "./src";

export default App;