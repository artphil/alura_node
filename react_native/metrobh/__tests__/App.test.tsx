import * as React from 'react';
import TestRenderer from "react-test-renderer"

import App from "../src"

describe("Componente App", () => {
  test("gera o mesmo Snapshot", () => {
    const tree: any = TestRenderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot();
  })
})