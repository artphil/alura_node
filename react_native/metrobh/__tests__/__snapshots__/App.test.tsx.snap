// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente App gera o mesmo Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.tsx to start working on your app!
  </Text>
</View>
`;
